python/cuml/cuml/manifold/spectral_embedding.pyx

```diff
@@ -0,0 +1,134 @@
+#
+# Copyright (c) 2025, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
+
+import cupy as cp
+import numpy as np
+
+from cython.operator cimport dereference as deref
+from libc.stdint cimport uint64_t, uintptr_t
+
+from pylibraft.common import cai_wrapper, device_ndarray
+from pylibraft.common.handle import auto_sync_handle
+
+from libcpp cimport bool
+from pylibraft.common.cpp.mdspan cimport (
+    col_major,
+    device_matrix_view,
+    make_device_matrix_view,
+    row_major,
+)
+from pylibraft.common.handle cimport device_resources
+
+from cuml.common import input_to_cuml_array
+from cuml.internals.base import Base
+from cuml.internals.mixins import CMajorInputTagMixin, SparseInputTagMixin
+
+
+cdef extern from "cuvs/preprocessing/spectral_embedding.hpp" namespace \
+        "cuvs::preprocessing::spectral_embedding" nogil:
+    cdef cppclass params:
+        int n_components
+        int n_neighbors
+        bool norm_laplacian
+        bool drop_first
+        uint64_t seed
+
+cdef params config
+
+cdef extern from "cuml/manifold/spectral_embedding.hpp" namespace "ML::SpectralEmbedding":
+
+    cdef int spectral_embedding_cuvs(
+        const device_resources &handle,
+        params config,
+        device_matrix_view[float, int, row_major] dataset,
+        device_matrix_view[float, int, col_major] embedding) except +
+
+
+@auto_sync_handle
+def spectral_embedding(A,
+                       n_components,
+                       random_state=None,
+                       n_neighbors=None,
+                       norm_laplacian=True,
+                       drop_first=True,
+                       handle=None):
+
+    cdef device_resources *h = <device_resources*><size_t>handle.getHandle()
+
+    A, _n_rows, _n_cols, _ = \
+        input_to_cuml_array(A, order="C", check_dtype=np.float32,
+                            convert_to_dtype=cp.float32)
+    A_ptr = <uintptr_t>A.ptr
+
+    config.n_components = n_components
+    config.seed = random_state if random_state is not None else 42
```

Comment:
We don't want to use a hardcoded seed of `42` here. Please use `cuml.internals.utils.check_random_seed` to coerce this to a valid random seed the same way other algorithms do:

```suggestion
    config.seed = check_random_seed(random_state)
```